### DISCUSSION TITLE
Valor do pedido em card token para assinaturas

### DISCUSSION THREAD

**Discussion by lucaslangsch** [mercadopago/sdk-js#208] [Q&A] [answered]
Um dúvida pensando na rejeição das operadores de cartão. Ao gerar um card token para ser usado na criação de assinaturas, qual o valor eu devo solicitar na requisição?

Exemplo, um plano de 12 meses com cobrança mensal no valor de 100,00. Qual deve ser o amount para gerar um token no sdk-js? 100,00 ou 1200,00?

`const cardForm = mp.cardForm({
  amount: XX,
  iframe: true,
  form: {
    id: "form-checkout",
  },
  callbacks: {
    onFormMounted: () => {},
    onSubmit: () => {},
    onFetching: () => {}
  },
});`

**Comment by lucaslangsch** [comment]
Acredito que faça mais sentido gerar o token com o valor da cobrança mensal, neste caso 100,00.

O Mercado Pago que vai tratar das cobranças futuras ou eu quem devo gerar um novo token a cada mês?

**Comment by lemazzo** [comment, marked as answer]
Oi, Lucas! 

Na criação de assinaturas cobra-se o valor mensal presente no plano ou na própria assinatura. No caso que usou de exemplo, o valor de amount seria 100,00.

Então, o fluxo é o seguinte:
- Front-end: Usa a SDK-JS com [Card Payment Brick](https://www.mercadopago.com.br/developers/pt/docs/checkout-bricks/card-payment-brick/introduction) ou [Payment Brick](https://www.mercadopago.com.br/developers/pt/docs/checkout-bricks/payment-brick/introduction) para gerar um token;
- Front-end: Envia esse token para o back-end (junto com outras informações relevantes obtidas no front-end);
- Back-end: Usa os dados obtidos para criar uma assinatura. Para esse passo você pode usar  a requisição de [criação de assinatura](https://www.mercadopago.com.br/developers/pt/reference/subscriptions/_preapproval/post) ou alguma SDK de back-end do Mercado Pago.

Caso tenha dúvidas na sua integração, você também pode abrir um ticket a partir [dessa página](https://www.mercadopago.com.br/developers/pt/support/30984). Espero que tenha ajudado

**Reply by lucaslangsch** [reply]
Entendi, e quanto aos próximos vencimentos, quem lida com isso?

Se eu configurar o objeto 'auto_recurring' o Mercado Pago se encarrega das cobranças futuras? Ou eu quem devo salvar o cliente e seu cartão e fazer o débito todo mês?

**Comment by lemazzo** [comment]
Bom dia, Lucas!

Sim, o Mercado Pago se encarrega das cobranças futuras. Essa é justamente uma grande vantagem de utilizar assinaturas!
